_posts/2020-06-04-spring-boot-hazelcast.md

```diff
@@ -304,6 +304,115 @@ Only after the data has been evicted from the near-cache will it be read from th
 
 **We should use the near-cache when we read from the cache very often, and when the data in the cache cluster changes only rarely.**
 
+## Serialization
+The java objects, that are stored in the cache should be serialized. 
+The `Car` class implements `Serializable`. In this case, Hazelcast will use
+the usual java serialization.
+
+But the standard java serialization has drawbacks like usage of more time and CPU, more space allocation.
+
+Let's imagine we have a scalable system with multiple instances and a cache cluster with few members.
+The system is working and cache entries are being stored, read, and evicted from the cache.
+Now we want to change a java object, that is cached and often used. We need a new deployment with this new class,
+and we want to make it without downtime. If we start a rolling update of our application instances,
+it works fine for the application, but the cache still can have entries of the previous version. In this case,
+Hazelcast will not be able to deserialize these entries. 
```

Comment:
I guess Hazelcast won't crash, but instead just produce cache misses, right? I added that, please correct it if it's wrong.

Reply:
I tested it. The serialization crashes, id we don't change the `key`, because the `value` is found in the cache, but this value cannot be deserialized. We get the `cache miss` if we change the `key`.